src/server/api/routers/recipes/resipes.ts

```diff
@@ -1,60 +1,57 @@
-import { eq, ilike, or } from "drizzle-orm";
+import { eq, ilike, or, sql } from "drizzle-orm";
 import { createTRPCRouter, publicProcedure } from "../../trpc";
 import { recipes, recipesToIngredients } from "~/server/db/schema";
 import { z } from "zod";
 
 export const recipesRouter = createTRPCRouter({
   getAllRecipes: publicProcedure
-    .input(z.object({ search: z.string(), sortBy: z.string().optional() }))
+    .input(
+      z.object({
+        search: z.string(),
+        sortBy: z.string(),
+        elements: z.number(),
+        page: z.number(),
+      }),
+    )
     .query(async ({ ctx, input }) => {
-      switch (input.sortBy) {
-        case "createdAt":
-          return ctx.db.query.recipes.findMany({
-            orderBy: (recipes, { desc }) => [desc(recipes.createdAt)],
-            with: {
-              recipesToIngredients: {
-                with: {
-                  ingredient: true,
-                },
-              },
-            },
-            where: or(
-              ilike(recipes.name, `%${input.search}%`),
-              ilike(recipes.shortDescription, `%${input.search}%`),
-            ),
-          });
+      const countResult = await ctx.db
+        .select({ count: sql<number>`count(*)` })
+        .from(recipes)
+        .where(
+          or(
+            ilike(recipes.name, `%${input.search}%`),
+            ilike(recipes.shortDescription, `%${input.search}%`),
+          ),
+        );
 
-        case "name":
-          return ctx.db.query.recipes.findMany({
-            orderBy: (recipes, { desc }) => [desc(recipes.name)],
-            with: {
-              recipesToIngredients: {
-                with: {
-                  ingredient: true,
-                },
-              },
-            },
-            where: or(
-              ilike(recipes.name, `%${input.search}%`),
-              ilike(recipes.shortDescription, `%${input.search}%`),
-            ),
-          });
+      const count = countResult[0]?.count || 0;
 
-        default:
-          return ctx.db.query.recipes.findMany({
+      const data = await ctx.db.query.recipes.findMany({
+        limit: input.elements,
+        offset: (input.page - 1) * input.elements,
+        orderBy: (recipes, { desc }) => [
+          desc(recipes[input.sortBy as keyof typeof recipes]),
```

Comment:
validate this for example:
const sortBySchtma = z.enum(['name', 'createdAt' ...])

then parse with safeparse

const isParsed = sortBySchtma.safeparse(input.sortBy)
if (isParsed.success) {
return....
} else {
set some default value
}

Reply:
or you can type it with zod schema then infer type and set it for your sortBy in zustand